sympy/printing/tests/test_str.py

```diff
@@ -809,3 +810,9 @@ def test_MatrixSymbol_printing():
 def test_Subs_printing():
     assert str(Subs(x, (x,), (1,))) == 'Subs(x, x, 1)'
     assert str(Subs(x + y, (x, y), (1, 2))) == 'Subs(x + y, (x, y), (1, 2))'
+
+def test_issue_15716():
+    x = Symbol('x')
+    e = -3**x*exp(-3)*log(3**x*exp(-3)/factorial(x))/factorial(x)
+    definite = integrate(e, [x, -oo, oo])
+    assert str(Integral(e, (x, -oo, oo)).doit()) == str(definite)
```

Comment:
The integral should be computed only once to save time. `Integral(e, (x, -oo, oo)).doit()` will only repeat what `integrate` does on the previous line. Perhaps it would be possible to write something like

    assert str(integrate(e, (x, -oo, oo))) == "..."

where the string on the right hand side is precomputed to save time.

Reply:
okk